homeassistant/components/tuya/base.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+"""Tuya Home Assistant Base Device Model."""
+from __future__ import annotations
+
+import asyncio
+
+from tuya_iot import TuyaDevice, TuyaDeviceManager
+
+from .const import DOMAIN
+
+
+class TuyaHaDevice:
+    """Tuya base device."""
+
+    def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager):
+        """Init TuyaHaDevice."""
+        super().__init__()
+
+        self.tuya_device = device
+        self.tuya_device_manager = device_manager
+
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
+
+    @staticmethod
+    def remap(old_value, old_min, old_max, new_min, new_max):
+        """Remap old_value to new_value."""
+        new_value = ((old_value - old_min) / (old_max - old_min)) * (
+            new_max - new_min
+        ) + new_min
+        return new_value
+
+    @property
+    def should_poll(self) -> bool:
+        """Hass should not poll."""
+        return False
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique ID."""
+        return f"tuya.{self.tuya_device.id}"
```

Comment:
No need to prefix unique IDs with your domain. It's only checked for uniqueness among other entities within same integration + platform (ie tuya switch)

Reply:
The old tuya integration unique_id is f"tuya.{self._tuya.object_id()}, so I think prefix is necessary.
<img width="596" alt="WeChatWorkScreenshot_7582de2f-67d3-4dc6-8eb5-02b9ae3a7fa0" src="https://user-images.githubusercontent.com/18442047/127090229-742acd78-da58-4235-bd95-07d917bb112b.png">

Reply:
Ok!